uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/request-modal-details-toil.controller.js

```diff
@@ -262,7 +264,8 @@ define([
      * @return {Promise}
      */
     function onDateChangeExtended () {
-      return calculateToilExpiryDate().catch($q.resolve);
+      return (canCalculateExpiryDate() && calculateToilExpiryDate().catch($q.resolve)) ||
+        $q.resolve();
```

Comment:
Easier to read

```
return canCalculateExpiryDate() ?
  calculateToilExpiryDate().catch($q.resolve)) :
  $q.resolve();
```

Reply:
Thanks for the suggestion! The linter likes it like this though:

```js
return canCalculateExpiryDate()
  ? calculateToilExpiryDate().catch($q.resolve)
  : $q.resolve();
```